### DISCUSSION TITLE
Can we use the new Microsoft Search SharePoint and OneDrive internal content source in the PnP Search Results web part?

### DISCUSSION THREAD

**Discussion by renwein** [microsoft-search/pnp-modern-search#2725] [Q&A] [answered]
Hi All - 

We have been testing the new SharePoint and OneDrive internal content source in Microsoft Search in the hopes of using some custom result types with the PnP Search Results web part on our company intranet.  We created our new result types a week ago.  

- [Microsoft Search: Custom result types in Microsoft Search](https://www.microsoft.com/en-us/microsoft-365/roadmap?filters=&searchterms=81952)
- [Manage result types](https://learn.microsoft.com/en-us/microsoftsearch/manage-result-types)
- [Manage result layouts for SharePoint results in Microsoft Search](https://vladilen.com/software/microsoft-365/manage-result-layouts-for-sharepoint-results-in-microsoft-search/)

We are using v4.8.0, and the issue that we're running into is that the web part setting asks for "IDs of connections defined in the Microsoft Search connectors administration portal" (see image).  The internal content source does not automatically generate a visible connection or corresponding ID in the admin center that we can use, and we don't seem to be able to create the connection because we're not using an external content source.  

We also get the error message fron the web part stating "Content source must be of the format - '/external/connections/connection_name'," but again since no visible connection is available in the admin center, we don't know what the formal connection name is to use, either.  

Does anyone know if there is a way to use the new internal SharePoint and OneDrive content source in the Search Results web part now, or is that currently unsupported until such time that it may be incorporated into a future release?  I suspect that the answer is the latter, but if anyone has any suggestions or ideas, they will be much appreciated.  

![Connector ID](https://user-images.githubusercontent.com/123119748/215511077-8874814e-1c96-43f1-93e5-f710b08759a8.png)

**Comment by wobba** [comment]
For SharePoint items use driveItem/listItem as the entity, not external. External is for Graph Connector items.

**Reply by renwein** [reply]
Thank you for the fast response @wobba.  This is our first foray into Microsoft Search, and we were following the Github instructions that said to use external for the entity to enable result types.  List items would definitely make more sense.  

The "Enable result types" slider gets grayed out when we switch the entity to List Items, so are you able to advise how can we then connect the web part to the Microsoft Search results types that we've created?  They contain our custom formatting, field mapping, and rules, which is what we would like to display in the web part if possible.  

![Result types disabled](https://user-images.githubusercontent.com/123119748/215568665-beb18188-1a56-4abb-a1e0-98c245d028b8.png)

**Reply by wobba** [reply]
Result types for SharePoint items are still in preview, so docs are still correct :) And we will enable it in the web parts when in released. In the mean time you need to do a code fix yourself to enable it.

**Comment by kasperbolarsen** [comment, marked as answer]
@renwein you are on the bleeding edge. The ability to create Result Types for ListItems and so forth is in public preview right now.

**Comment by renwein** [comment]
Yeah, we saw the new feature, got excited, and wanted to run with it as far as we could.  Thank you so much @wobba and @kasperbolarsen for taking the time to confirm that we're ahead of ourselves at the moment.

**Reply by kasperbolarsen** [reply]
No problem 🙂

Actually I am working on an update on the Microsoft search section, and you can use that Data Source, just not the Result Type feature.
